### PR TITLE
Updating What's new; adding support for Mermaid (diagrams); update debug info

### PR DESCRIPTION
**Pull request by d45**
@johnDance @seanmakesgames @shinchris @KeshiaRose 
Doc update (belated) for v1.5 library
Changes in this PR:

* Updated "What's New" with links to v1.5 release notes
* Updated debugging desktop instructions for 2021.1 and later (no **Pause before loading**)
* Added sequence diagram for the OAuth flow (using Mermaid gem ) [https://mermaid-js.github.io/](https://mermaid-js.github.io/) 
* Updated `head.html` file to initialize Mermaid
* Updated `_config.yml` to add memaid gem

### CODE REVIEW

File: docs/trex_release-notes.md [hunk 1]
```diff
@@ -11,9 +11,44 @@ layout: docs
...
+
+About this release:
+
+* The `Filter.getFieldAsync` method now works as expected and properly returns the field. This fix requires Tableau 2019.2 and later.
+
+* Show/Hide (`setZoneVisibilityAsync`) can now be applied to any dashboard zone.
+
+* The `selectMarksByValueAsync` method now supports combined selection criteria types (bug fixed).
+
+* The following are all improvements to the `getSummaryData` method:
```
Comment (johnDance) [inline docs/trex_release-notes.md:30]: For user documentation, we should always spell out getSummaryDataAsync, so they know the exact method. (Change all from getSummaryData to getSummaryDataAsync)

File: docs/trex_release-notes.md [hunk 2]
```diff
@@ -11,9 +11,44 @@ layout: docs
...
+
+  * `getSummaryData` now has a smaller and faster payload.
+
+  * `maxRows` can be applied to `getSummaryData` to restrict the number of rows fetched.
+
+  * `columnsToIncludeById` can be applied to `getSummaryData` to restrict the columns fetched.
+
+  * `includeDataValuesOnly` or `includeFormattedValuesOnly` can be applied to `getSummaryData` to restrict the amount of information returned to what you really need.
+
+  * The `getSummaryColumnsInfoAsync` method is a new in this release. It returns the column information for each column that will be returned in `getSummaryData`.
```
Comment (johnDance) [inline docs/trex_release-notes.md:40]: "method is a new in this release". There is a word or two missing, or just in the wrong order.

File: docs/trex_release-notes.md (hunk 1 above)
Comment (d45) [reply docs/trex_release-notes.md:30]: Yikes! Thanks for catching that, John. I'll fix all the instances and add a link to the reference page.

File: docs/trex_release-notes.md (hunk 2 above)
Comment (d45) [reply docs/trex_release-notes.md:40]: yes. There are words missing there. Thanks for catching that.   I think it was supposed to say:


The `getSummaryColumnsInfoAsync` method is new in this release. It returns the column information for each column that will be returned in `getSummaryDataAsync`.

File: docs/trex_manifest.md [hunk 3]
```diff
@@ -13,18 +13,21 @@ For details about a manifest or its fields, see the [Sample Manifest File](#samp
...
 ## Manifest Versioning
+
 The versioning of the manifest is designed to be semantically simple and support compatibility. The version follows the [Major].[Minor] format. Minor upgrades are backwards compatible while major upgrades involve breaking changes. 
 
 ## Error Reporting
+
 At start up, Tableau checks the manifest file. If any errors are found while parsing the file, Tableau writes these errors to the `log.txt` file in the `My Tableau Repository (Beta)/Logs` folder. This is the same location that Tableau Desktop uses to report other errors and activity. 
 
-If a workbook is saved with an extension and then later opened on another computer that does not have the extension installed, Tableau displays a message in the dashboard zone where the extension would have appeared that states the extension is not available. 
+If a workbook is saved with an extension and then later opened on another computer that does not have the extension installed, Tableau displays a message in the dashboard zone where the extension would have appeared that states the extension is not available.
```
Comment (johnDance) [inline docs/trex_manifest.md:30]: This message is confusing. What do we mean "another computer that does not have the extension installed"?

Comment (d45) [reply docs/trex_manifest.md:30]: Right.... This is pretty old. I think this is the case where the user/developer is sharing a dashboard extension that is expected to be running on the developer's localhost. Or perhaps it's from when the user had to have the physical `.trex` file in their local `My Tableau Repository` folder?  Either way, this is confusing.

Comment (johnDance) [reply docs/trex_manifest.md:30]: I agree. I vote to just delete it.

Comment (d45) [reply docs/trex_manifest.md:30]: I've also removed the `beta` from the path in the previous paragraph. 
"writes these errors to the `log.txt` file in the `My Tableau Repository (Beta)/Logs` folder. "

**Review by johnDance — APPROVED** [review]

**Pull request merged**